### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by julianvmodesto**
For pushing to `liveramp/elastalert` on Docker Hub.

### CODE REVIEW

**Comment by julianvmodesto** [comment]
Same as https://github.com/Yelp/elastalert/pull/1644

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM alpine
+
+RUN apk update && \
+    apk upgrade && \
```
Comment (joshk0) [inline Dockerfile]: calling `apk upgrade` is probably overkill

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    apk add ca-certificates openssl python2 py2-pip py2-yaml tzdata && \
+    apk add --virtual build-dependencies python2-dev musl-dev gcc openssl-dev libffi-dev
```
Comment (joshk0) [inline Dockerfile]: You should do an apk clean at the end of this layer, and move the build-dependency apk call further down.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+ENV ELASTALERT_HOME /opt/elastalert/
+
+WORKDIR /opt
+
+ADD setup.py requirements.txt elastalert "${ELASTALERT_HOME}"
+
+WORKDIR "${ELASTALERT_HOME}"
+
+RUN pip install -r requirements.txt && \
```
Comment (joshk0) [inline Dockerfile]: Here is where the second call to apk should be.

**Comment by bobisme** [comment]
what if the docker build were a separate repo until the dockerfile is merged into the upstream?
if they decide that they don't want this, then this fork will need to be maintained forever. and i'm all about not keeping forks around if changes can be merged into the upstream -- not sure if there's anything else preventing that.

**Comment by julianvmodesto** [comment]
I'll setup an automated Docker build off of this repo, so all we'd have to do is merge in upstream master and tags, and then it'll kick off an automated build.

idk it seems similar to have an empty project because if we do it that way, then we'd still have to manually push a tag to kick off an automated build. Either way, I'm not planning on us merging in custom changes to the actual Python module so I guess it kind of makes sense for an empty project. 🤷‍♂️

File: Dockerfile (hunk 1 above)
Comment (julianvmodesto) [reply Dockerfile]: Done.

File: Dockerfile (hunk 2 above)
Comment (julianvmodesto) [reply Dockerfile]: Looks like I cannot do a `apk cache clean`, I get
```
ERROR: Package cache is not enabled.
```

Do you mean this?
```
rm -f /var/cache/apk/*
```
Found from here: https://github.com/gliderlabs/docker-alpine/issues/224

File: Dockerfile (hunk 3 above)
Comment (julianvmodesto) [reply Dockerfile]: Done.

**Comment by julianvmodesto** [comment]
@joshk0 PTAL

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
+FROM alpine
+
+RUN apk update && \
+    apk add ca-certificates openssl python2 py2-pip py2-yaml tzdata
```
Comment (joshk0) [inline Dockerfile]: Use `apk add --no-cache`. `apk update` is not required

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+ENV ELASTALERT_HOME /opt/elastalert/
+
+WORKDIR /opt
+
+ADD setup.py requirements.txt elastalert "${ELASTALERT_HOME}"
+
+WORKDIR "${ELASTALERT_HOME}"
+
+RUN apk add --virtual build-dependencies python2-dev musl-dev gcc openssl-dev libffi-dev && \
```
Comment (joshk0) [inline Dockerfile]: also here

**Pull request merged**